src/packages/blits/src/components/Metadata/Details.ts

```diff
@@ -0,0 +1,237 @@
+import Blits from '@lightningjs/blits';
+import styles from './Details.styles';
+import Badge, { BadgeProps } from '../Badge/Badge';
+import { getStyledProp } from '../../utils/styleUtils';
+import { getTextWidth } from '../../utils/textUtils';
+import Rating, { RatingProps } from './Rating';
+import { getFlexRects, getMaxHeight, type FlexRect, isValidState, isValidTone } from '../../utils';
+import { States, Tone } from '../../types/types';
+
+export type DetailsProps = {
+  badges?: BadgeProps[];
+  ratings?: RatingProps[];
+  title?: string;
+};
+
+type DetailsState = {
+  titleProps: {
+    font: string;
+    size: number;
+    lineheight: number;
+  };
+  badgeProps: {
+    gap: number;
+    height: number;
+    rects: FlexRect[];
+    x: number;
+  };
+  ratingProps: {
+    gap: number;
+    height: number;
+    rects: FlexRect[];
+    x: number;
+  };
+  gap: number;
+};
+
+const Details = Blits.Component('Details', {
+  props: [
+    {
+      key: 'states',
+      default: 'focus',
+      required: false,
+      cast: (v: string): States => {
+        if (isValidState(v)) return v;
+        throw new Error(`Invalid state '${v}'`);
+      }
+    },
+    {
+      key: 'tone',
+      default: 'neutral',
+      required: false,
+      cast: (v: string): Tone => {
+        if (isValidTone(v)) return v;
+        throw new Error(`Invalid tone '${v}'`);
+      }
+    },
+    {
+      key: 'width',
+      required: false,
+      default: 770,
+      cast: Number
+    },
+    {
+      key: 'badges',
+      required: false,
+      cast: (v: unknown) => {
+        if (Array.isArray(v)) return v;
+        throw new Error('Badges must be an array');
+      }
+    },
+    {
+      key: 'ratings',
+      required: false,
+      cast: (v: unknown) => {
+        if (Array.isArray(v)) return v;
+        throw new Error('Ratings must be an array');
+      }
```

Comment:
Thanks for spotting big oversight from me, will get these fixed!

Reply:
Have been playing around with this a bit more and found that if you add `cast` to a prop, it will always run, whether the prop is required or not, and if nothing is passed it uses `undefined`.

The same issue you found with these props is also present when using the String/Number casters from the docs
```
{
  key: 'title',
  required: false,
  cast: String
}
```
So even if you didn't pass a `title` prop, it will always set itself to the string 'undefined'. It looks like `required: false` doesn't actually do anything, just `required: true` will add a console warning if the prop is missing.

I think this is all a side-effect of needing to use cast to validate types rather than the validate function that's in the docs but not in Blits yet.  I'll add null checks to the casts to fix this for now, but hopefully once validate is available we can move away from cast.